src/services/httpClient.js

```diff
@@ -76,14 +84,34 @@ export class HttpClient {
       .retryWhen(retryHandler(maxRetryAttempts, retryWaitTime));
   }
 
+  /**
+  * Constructs an Ajax request for a request that does not return json
+  *
+  * @param {string} url The url path to the make the request to
+  * @param {AjaxRequest} [options={}] See https://github.com/ReactiveX/rxjs/blob/master/src/observable/dom/AjaxObservable.ts
+  * @param {boolean} withAuth Allows a backdoor to not avoid wrapping auth headers
+  * @return an Observable of the AjaxReponse
+  */
+ static ajaxNonJson(url, options = {}, withAuth = true) {
+   const { retryWaitTime, maxRetryAttempts } = Config;
+   const request = HttpClient.createAjaxRequest({ ...options, url }, withAuth, false);
+   return Observable.ajax(request)
+     // Classify errors as retryable or not
+     .catch(ajaxError => Observable.throw(classifyError(ajaxError)))
+     // Retry any retryable errors
+     .retryWhen(retryHandler(maxRetryAttempts, retryWaitTime));
+ }
+
   /**
    * A helper method that adds "application/json" headers
    */
-  static withHeaders(request, withAuth) {
+  static withHeaders(request, withAuth, addJson = true) {
```

Comment:
You don't need all the extra changes in this file. I think changing only the withHeaders method should do the trick.

The httpClient should probably refactored some more to make these cases easier to handle but this should work alright with minimal pain and added complexity. When you don't want the json headers you can pass your own headers in the options and override the `Accept` and `Content-Type` to `undefined` or your own override value.

```
// Outside the class
const jsonHeaders = {
  'Accept': 'application/json',
  'Content-Type': 'application/json'
};

// Inside the class
  static withHeaders(request, withAuth) {
    const authHeaders = {};
    if (withAuth) {
      authHeaders['Csrf-Token'] = 'nocheck';
      AuthService.getAccessToken(accessToken => {
        if (accessToken) authHeaders['Authorization'] = `Bearer ${accessToken}`;
      });
    }
    return {
      ...request,
      headers: {
        ...jsonHeaders,
        ...request.header,
        ...authHeaders
      }
    };
  }
```

Reply:
I also need to change the ajax response method so it doesn't try to camel case a json response that isn't there